### PR TITLE
own up to the attrocities commited in set

### PR DESCRIPTION
**Pull request by deads2k**
demonstrates what a mess we have.

We should never add flags again.  This is needed to stop having a fake factory.

/assign @mfojtik

### CODE REVIEW

**Comment by deads2k** [comment]
@mfojtik do we have any tests on this?  I'm surprised I got green out of the gate.

**Comment by deads2k** [comment]
/hold

Needs tidying before merge

**Comment by deads2k** [comment]
tidying completed.  Ready for review

**Comment by mfojtik** [comment]
@deads2k can we just get rid of ResolveImage altogether and just call the patched resolve image? The factory function is exclusively used only by 'set' command.

**Comment by deads2k** [comment]
> @deads2k can we just get rid of ResolveImage altogether and just call the patched resolve image? The factory function is exclusively used only by 'set' command.

I don't think we should do that in this patch.  I think that removing methods like that should be done upstream first.  Otherwise we just end up carrying debt.

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_image.go [hunk 1]
```diff
@@ -117,7 +122,7 @@ func (o *ImageOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []st
 	o.ChangeCause = f.Command(cmd, false)
 	o.DryRun = cmdutil.GetDryRunFlag(cmd)
 	o.Output = cmdutil.GetFlagString(cmd, "output")
-	o.ResolveImage = f.ResolveImage
+	o.ResolveImage = resolveImageFactory(f, cmd)
```
Comment (mfojtik) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_image.go]: can we get rid of this field for set image and just call the function directly in Run() ? Also remove it from upstream factory as it is not used anywhere else.

**Comment by mfojtik** [comment]
@deads2k i'm fine with follow up (I can do it after this merges)... that shim was introduced by us by trying to plug our image resolution into upstream kubectl. 

/lgtm

**Pull request merged**